src/garage/_dtypes.py

```diff
@@ -428,22 +437,42 @@ def next_observations(self):
             ]))
 
 
+class StepType(enum.IntEnum):
+    """Defines the status of a `TimeStep` within a sequence.
+
+    Note that the last `TimeStep` in a sequence can either be TERMINAL or
+    TIMEOUT.
+
+    Suppose max_path_length = 5:
+    * A success sequence terminated at step 4 will look like:
+        FIRST, MID, MID, TERMINAL
+    * A success sequence terminated at step 5 will look like:
+        FIRST, MID, MID, MID, TERMINAL
+    * An unsuccessful sequence truncated by time limit will look like:
+        FIRST, MID, MID, MID, TIMEOUT
+    """
+    # Denotes the first `TimeStep` in a sequence.
+    FIRST = 0
```

Comment:
RESET? INITIAL?

Reply:
Many libraries use the FIRST, MID, LAST pattern (such as [`dm_control`](https://github.com/deepmind/dm_env/blob/master/dm_env/_environment.py#L71) and `tensorflow`). I would advocate for keeping FIRST and MID so the naming is consistent. 

We also provide `last` as a property for `TimeStep`, so `garage.TimeStep` can be used in a similar way like external TimeStep. Depends on user's need they can decide whether they want to differentiate `TERMINAL` and `TIMEOUT` or not.

Reply:
dm_control/dm_eng and tf_agents are written by approximately same group of people (really two groups of people who work together a lot) so it's unsurprising that they use the same naming convention. Basically, I don't think that's much evidence for a concensus.

FIRST and MID make sense in the context of LAST. Once you remove LAST and replace it with TERMINAL and TIMEOUT, the naming convention becomes more surprising.

We should strive to use what we think will be least surprising to users, not cargo-culting other libraries.

Reply:
Let's discuss this in standup maybe?

Reply:
So the main takeaway from that standup was that FIRST is probably a bad idea anyways, because those steps act very differently from the other steps, and almost all algorithms would want to ignore those steps.

Reply:
Since we decided `env.reset()` shouldn't return `TimeStep` (because then we would end up with timesteps that don't contain actions / aren't actually a step in the underlying MDP), I will remove `FIRST` from the enum.

Reply:
Thoughts gathered from standup: `FIRST` should denote the first MDP time step, which is returned from the first `step()`. In other words `step()` can return step type `FIRST`/`MID`/`TERMINAL`/`TIMEOUT`.